ext/imap/php_imap.stub.php

```diff
@@ -0,0 +1,354 @@
+
+/**
+ * @param resource $stream_id
+ */
+function imap_gc($stream_id, int $flags): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_expunge($stream_id): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_delete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_undelete($stream_id, int $msg_no, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_check($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_listscan($stream_id, string $ref, string $pattern, string $content) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_copy($stream_id, string $msglist, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_mail_move($stream_id, string $sequence, string $mailbox, int $options = 0): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_mail_compose(array $envelope, array $body) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_createmailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_renamemailbox($stream_id, string $old_name, string $new_name): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_deletemailbox($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_subscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_unsubscribe($stream_id, string $mailbox): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_append($stream_id, string $folder, string $message, string $options = '', string $internal_date = ''): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_ping($stream_id): bool {}
+
+/**
+ * @return string|false
+ */
+function imap_base64(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_qprint(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_8bit(string $text) {}
+
+/**
+ * @return string|false
+ */
+function imap_binary(string $text) {}
+
+
+function imap_utf8(string $mime_encoded_text): string {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_status($stream_id, string $mailbox, int $options) {}
+
+/**
+ * @param resource $stream_id
+ * @return object|false
+ */
+function imap_mailboxmsginfo($stream_id) {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_setflag_full($stream_id, string $sequence, string $flag, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ */
+function imap_clearflag_full($stream_id, string $sequence, string $flag, int $options = 0): bool {}
+
+/**
+ * @param resource $stream_id
+ * @return array|false
+ */
+function imap_sort($stream_id, int $criteria, int $reverse, int $options = 0, string $search_criteria ='', string $charset = '') {}
```

Comment:
Not sure, but I think this should be:
```suggestion
function imap_sort($stream_id, int $criteria, int $reverse, int $options = 0, string $search_criteria =UNKNOWN, string $charset = UNKNOWN) {}
```

Reply:
I may be misunderstanding when `UNKNOWN` is appropriate, but for $charset, isn't it just passing the string as given, or `NIL` to the `mail_sort` c function, in https://github.com/php/php-src/blob/master/ext/imap/php_imap.c#L3201 ?

Reply:
`UNKNOWN` has to be used, if there is no more specific default value. In this case, passing an empty string as `$charset` would not call `mail_sort()` with the same arguments as when skipping this argument, since `""` is not equal to `NIL`. Both may be handled identically in `mail_sort`, so I'm still not sure, though. But if in doubt, `UNKNOWN` is preferable, and could still be revised later on.

Reply:
ah ok, thanks for the clarification.